specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2021-10-01/imagebuilder.json

```diff
@@ -0,0 +1,1553 @@
+              "title": "The operation type.",
+              "description": "For example: read, write, delete, or listKeys/action",
+              "type": "string"
+            },
+            "resource": {
+              "title": "The resource type on which the operation is performed.",
+              "type": "string"
+            },
+            "description": {
+              "title": "The friendly name of the operation",
+              "type": "string"
+            }
+          }
+        },
+        "origin": {
+          "title": "The intended executor of the operation.",
+          "type": "string"
+        },
+        "properties": {
+          "title": "Properties of the operation.",
+          "type": "object"
+        },
+        "isDataAction": {
+          "title": "The flag that indicates whether the operation applies to data plane.",
+          "type": "boolean"
+        }
+      }
+    },
+    "OperationListResult": {
+      "type": "object",
+      "title": "Result of the request to list REST API operations. It contains a list of operations and a URL nextLink to get the next set of results.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Operation"
+          },
+          "title": "The list of operations supported by the resource provider."
+        },
+        "nextLink": {
+          "type": "string",
+          "title": "The URL to get the next set of operation list results if there are any."
+        }
+      }
+    },
+    "CloudError": {
+      "type": "object",
+      "x-ms-external": true,
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/CloudErrorBody",
+          "description": "Details about the error."
+        },
+        "systemData": {
+          "$ref": "#/definitions/systemData",
+          "readOnly": true
+        }
+      },
+      "description": "An error response from the Azure VM Image Builder service."
+    },
+    "CloudErrorBody": {
+      "type": "object",
+      "x-ms-external": true,
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "An identifier for the error. Codes are invariant and are intended to be consumed programmatically."
+        },
+        "message": {
+          "type": "string",
+          "description": "A message describing the error, intended to be suitable for display in a user interface."
+        },
+        "target": {
+          "type": "string",
+          "description": "The target of the particular error. For example, the name of the property in error."
+        },
+        "details": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CloudErrorBody"
+          },
+          "description": "A list of additional details about the error."
+        }
+      },
+      "description": "An error response from the Azure VM Image Builder service."
+    },
+    "systemData": {
```

Comment:
>systemData [](http://example.com/codeflow?start=5&length=10)

Recommend pulling this from common-types.

Reply:
Hi @mentat9, can you elaborate?

Reply:
I actually did try referencing the systemData definition in the common-types folder but I kept getting an "Could not read" and "file not found" error so I decided to define it locally. 

ex:
"systemData": {
          "$ref": "./../../../../common-types/resource-management/v2/types.json#/definitions/systemData",
          "description": "Metadata pertaining to creation and last modification of the resource."
        }

Reply:
You have an invalid relative path, see here: [https://github.com/Azure/azure-rest-api-specs/blob/27c89d071d6a3d6edc908d86655773e0aa9e59d0/specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/deploymentScripts.json#L454](https://github.com/Azure/azure-rest-api-specs/blob/27c89d071d6a3d6edc908d86655773e0aa9e59d0/specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/deploymentScripts.json#L454)